src/CDN/CloudFrontVersion3.php

```diff
@@ -116,13 +116,22 @@ public function flushPaths(array $paths): string
                     'CallerReference' => $this->getCallerReference(),
                 ],
             ]);
-            $status = $result->get('Invalidation')['Status'];
+            $invalidation = $result->get('Invalidation');
+            \assert(\is_array($invalidation));
 
-            if (false === array_search($status, self::AVAILABLE_STATUSES, true)) {
-                throw new \RuntimeException(sprintf('Unable to determine the flush status from the given response: "%s".', $status));
+            $status = $invalidation['Status'] ?? null;
+
+            if (null === $status || !\in_array($status, self::AVAILABLE_STATUSES, true)) {
+                throw new \RuntimeException('Unable to determine the flush status from the given response.');
```

Comment:
Why getting rid of the status? looks like it's useful information

Reply:
Re add that exception (I split when having null as status from the not in the array of available statuses